src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.spec.ts

```diff
@@ -1,38 +1,15 @@
-import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { RouterTestingModule } from '@angular/router/testing';
 
-import { CollapseModule, PopoverModule } from 'ngx-bootstrap';
-
-import { NotificationService } from '../../../shared/services/notification.service';
-import { SharedModule } from '../../../shared/shared.module';
+import { AppModule } from '../../../app.module';
 import { configureTestBed } from '../../../shared/unit-test-helper';
-import { LogoutComponent } from '../../auth/logout/logout.component';
-import { NotificationsComponent } from '../notifications/notifications.component';
-import { TaskManagerComponent } from '../task-manager/task-manager.component';
 import { NavigationComponent } from './navigation.component';
 
 describe('NavigationComponent', () => {
   let component: NavigationComponent;
   let fixture: ComponentFixture<NavigationComponent>;
 
-  const fakeService = new NotificationService(null, null);
-
   configureTestBed({
-    imports: [
-      SharedModule,
-      RouterTestingModule,
-      HttpClientTestingModule,
-      CollapseModule.forRoot(),
-      PopoverModule.forRoot()
-    ],
-    declarations: [
-      NavigationComponent,
-      NotificationsComponent,
-      LogoutComponent,
-      TaskManagerComponent
-    ],
-    providers: [{ provide: NotificationService, useValue: fakeService }]
+    imports: [AppModule]
```

Comment:
Isn't this too generic? Shouldn't we at least declare the component we are testing and then import the necessary modules?

Reply:
The problem was that the navigation depends on nearly every module there is, I could add 20 dependency or just use `AppModule` which shows the reader of the test that this module depends on nearly everything.